CMakeLists.txt

```diff
@@ -55,7 +56,6 @@ set(PYBIND11_HEADERS
   include/pybind11/pytypes.h
   include/pybind11/stl.h
   include/pybind11/stl_bind.h
-  include/pybind11/typeid.h
```

Comment:
`typeid.h` does provide the non-`detail` namespace `py::type_id<T>()`, which is a fairly useful function and *might* be in use.  Maybe this one should be left for later?